model/src/guild/premium_tier.rs

```diff
@@ -1,14 +1,38 @@
-use serde_repr::{Deserialize_repr, Serialize_repr};
+use serde::{Deserialize, Serialize};
 
-#[derive(
-    Clone, Copy, Debug, Deserialize_repr, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize_repr,
-)]
-#[repr(u8)]
+
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize)]
+#[serde(from = "u8", into = "u8")]
 pub enum PremiumTier {
-    None = 0,
-    Tier1 = 1,
-    Tier2 = 2,
-    Tier3 = 3,
+    None,
+    Tier1,
+    Tier2,
+    Tier3,
+    Other(u8)
```

Comment:
Similar to `AutoArchiveDuration`, I'm not sure why this is Other and not Unknown.

Reply:
This one i think it's a safe assumption that discord won't be adding a 2.5 as number 4. So we know what the value will represent if they add a new one, the new tier